textile/chat-features.textile

```diff
@@ -411,36 +411,56 @@ Presence allows chat room users to indicate to others that they're online, as we
+* @(CHA-T2)@ This specification point has been removed. It was replaced by @CHA-T9@.
+* @(CHA-T8)@ Typing Indicators for a Room are exposed on the main room channel, in the format @<roomId>::$chat@. For example, if your room id is @my-room@ then the channel will be @my-room::$chat@.
+* @(CHA-T9)@ @[Testable]@ Users may retrieve a list of the currently typing client IDs.
+* @(CHA-T10)@ @[Testable]@ Users may configure a heartbeat interval (the no-op period for @typing.keystroke@ when the heartbeat timer is set active at @CHA-T4a4@). This configuration is provided at the @RoomOptions.typing.heartbeatThrottleMs@ property, or idiomatic equivalent. The default is 10000ms.
+** @(CHA-T10a)@ @[Testable]@ A grace period shall be set by the client (the grace period on the @CHA-T10@ heartbeat interval when receiving heartbeats). The value shall be set to 2000ms.
+*** @(CHA-T10a1)@ @[Testable]@ If a @typing.start@ event is not received within this period, the client shall assume that the user has stopped typing. For example, if the client has not received a @typing.start@ event within 12000ms of the last heartbeat (10s heartbeat interval plus 2s grace period), the client shall assume that the user has stopped typing.
+* @(CHA-T3)@ This specification point has been removed. It was replaced by @CHA-T10@.
+* @(CHA-T4)@ Users may indicate that they have started typing using the @keystroke@ method.
+** @(CHA-T4d)@ @[Testable]@ If the channel state is not @ATTACHED@ or @ATTACHING@, the operation shall throw an @ErrorInfo@ with code @40000@.
+** @(CHA-T4a)@ If typing is not already in progress (i.e. a heartbeat timer not set on successful publish @CHA-T4a4@):
+*** @(CHA-T4a1)@ This specification point has been removed. It was replaced by @CHA-T4a3+@.
+*** @(CHA-T4a2)@ This specification point has been removed. It was replaced by @CHA-T4a3+@.
+*** @(CHA-T4a3)@ @[Testable]@ The client shall publish an ephemeral message to the channel with the name field set to @typing.started@, the format of which is detailed "here":#realtime-typing-heartbeat.
+*** @(CHA-T4a4)@ @[Testable]@ Upon successful publish, a heartbeat timer shall be set according to the @CHA-T10@ timeout interval.
+** @(CHA-T4b)@ This specification point has been removed. It was superseded by @CHA-T4c@.
+** @(CHA-T4c)@ If typing is already in progress (i.e. a heartbeat timer set according to @CHA-T4a4@ exists and has not expired):
+*** @(CHA-T4c1)@ @[Testable]@ The client must not send a @typing.started@ event.
+* @(CHA-T5)@ Users may explicitly indicate that they have stopped typing using @stop@ method.
+** @(CHA-T5a)@ @[Testable]@ If typing is not in progress (i.e. a @CHA-T10@ heartbeat timer does not exist or is expired), this operation is a no-op.
+** @(CHA-T5b)@ This specification point has been removed.
+** @(CHA-T5c)@ @[Testable]@ If the channel state is not @ATTACHED@ or @ATTACHING@, the operation shall throw an @ErrorInfo@ with code @40000@.
```

Comment:
@(CHA-T5c) will be updated same as https://github.com/ably/specification/pull/277/files#r2019765235.
For now, we can ignore this.